packages/preview-middleware/src/base/flp.ts

```diff
@@ -403,6 +415,56 @@ export class FlpSandbox {
         }) as RequestHandler);
     }
 
+    /**
+     * Add routes for html and scripts required for a local test FLP.
+     *
+     * @param configs test configurations
+     * @param id application id from manifest
+     */
+    private addTestRoutes(configs: TestConfig[], id: string) {
+        const ns = id.replace(/\./g, '/');
+        const htmlTemplate = readFileSync(join(__dirname, '../../templates/test/qunit.html'), 'utf-8');
+        const initTemplate = readFileSync(join(__dirname, '../../templates/test/qunit.js'), 'utf-8');
+        for (const testConfig of configs) {
+            const config = mergeTestConfigDefaults(testConfig);
+            this.logger.debug(`Add route for ${config.path}`);
+            // add route for the *.qunit.html
+            this.router.get(config.path, (async (_req, res, next) => {
+                this.logger.debug(`Serving test route: ${config.path}`);
+                const file = await this.project.byPath(config.path);
+                if (file) {
+                    this.logger.warn(`HTML file returned at ${config.path} is loaded from the file system.`);
+                    next();
+                } else {
+                    const templateConfig = {
+                        id,
+                        framework: config.framework,
+                        basePath: posix.relative(posix.dirname(config.path), '/') ?? '.',
+                        initPath: `${ns}${config.init.replace('.js', '')}`
+                    };
+                    const html = render(htmlTemplate, templateConfig);
+                    res.status(200).contentType('html').send(html);
+                }
+            }) as RequestHandler);
+            // add route for the init file
+            this.logger.debug(`Add route for ${config.init}`);
+            this.router.get(config.init, (async (_req, res, next) => {
```

Comment:
@tobiasqueck In case the test config has a custom `init: /test/integration/opaTests.qunit` then a route is being created to `/test/integration/opaTests.qunit` that is never used because the file `/test/integration/opaTests.qunit.js` is being loaded from the file system. Nevertheless there's also this log message `debug preview-middleware Add route for /test/integration/opaTests.qunit` that is kind of misleading. Should we skip the whole thing in case a custom `init` is given (my preference) or enhance the custom init by `.js` so that the route can do the skip (line 456)?

Reply:
I see you decided already and added
```js
if (testConfig.init !== undefined) {
       this.logger.debug(`Skip serving test init script in favor of provided script: ${testConfig.init}`);
        continue;
}
```
Good with me :)